### PR TITLE
refactor: make tilt friendlier

### LINKED ISSUE #762: Introducing runtime namespace env variable and downward map Pod namespace to the variable
**User Story:**
As a Kubernetes cluster administrator, I want to introduce a new environment variable, RUNTIME_NAMESPACE, that will map the Pod's namespace at runtime so that my microservices can be more aware of their deployed environment.

**Acceptance Criteria:**
  - The system should introduce a new environment variable named RUNTIME_NAMESPACE.
  - The RUNTIME_NAMESPACE environment variable should automatically map the namespace of the Pod at runtime.
  - The value of RUNTIME_NAMESPACE should be dynamically derived from the Pod's `metadata.namespace` field.

```yaml
        env:
        - name: RUNTIME_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
```

### LINKED ISSUE #770: Better Tilt support, `tilt up` would work out of the box for the branch planner

### PR DESCRIPTION
**Pull request by yitsushi**
Previously we had a bunch of yaml files under `config/`, sometimes they
become outdated because no one updated them. With the new `Tiltfile`, on
`tilt up` it will use the helm chart from `charts/tf-controller`.

What it does:

1. Builds the controller image.
2. Builds the branch planner image.
3. Builds the runner image.
4. Update Helm chart dependencies.
5. Deploys the helm chart with `config/tilt/helm/dev-values.yaml`.
5. Creates a `terraform` namespace.
6. Creates a `ConfigMap` to watch `Terraform` objects in the `terraform`
   namespace.
7. Creates a `Secret` from the `GITHUB_TOKEN` environment variable.

If there is a `Tiltfile.local`, it will source that file too. This file
is ignored by git, if you need to add things specific to your local
environment, you can add those in that file.

If there is a `config/tilt/helm/dev-values-local.yaml` file, it will use
that file to feed values to the helm deployment as an addition to
`config/tilt/helm/dev-values.yaml`.

Closes #770

## Additional changes

**Add `RUNTIME_NAMESPACE` to branch planner**

```
❯ kubectl describe pod -n flux-system chart-tf-controller-planner-6858b948fc-qrdch | grep RUNTIME_NAMESPACE
      RUNTIME_NAMESPACE:  flux-system (v1:metadata.namespace)
```

Closes #762

**Add build tags to dockerfiles**

None of the dockerfiles used `BUILD_VERSION` and `BUILD_SHA` to build
the go application. Now they do.

References:
* https://github.com/weaveworks/tf-controller/issues/770
* https://github.com/weaveworks/tf-controller/issues/762

Signed-off-by: Balazs Nadasdi <balazs@weave.works>

### CODE REVIEW

**Comment by yitsushi** [comment]
image

```
❯ kubectl get terraforms.infra.contrib.fluxcd.io -A
NAMESPACE   NAME                              READY     STATUS                                                  AGE
terraform   helloworld-tf                     Unknown   Plan generated: This object is in the plan only mode.   100s
terraform   helloworld-tf-chanwit-patch-1-7   Unknown   Initializing                                            17s
```

**Comment by yitsushi** [comment]
Now `helloworld-tf` is listed under `resources`, that contains all runner logs (for the main tf object and all branch planner clones)
image

File: Tiltfile [hunk 1]
```diff
@@ -1,35 +1,89 @@
...
+load('ext://secret', 'secret_from_dict')
+load('ext://namespace', 'namespace_create', 'namespace_inject')
 
-k8s_yaml(kustomize('config/tilt/manager'))
-k8s_yaml(kustomize('config/tilt/branch-planner'))
+namespace       = "flux-system"
+tfNamespace     = "terraform"
+buildSHA        = str(local('git rev-parse --short HEAD')).rstrip('\n')
+buildVersionRef = str(local('git rev-list --tags --max-count=1')).rstrip('\n')
+buildVersion    = str(local("git describe --tags ${buildVersionRef}")).rstrip('\n')
```
Comment (yiannistri) [inline Tiltfile:10]: ❤️

**Review by yiannistri — APPROVED** [review]
Tested and works great 👌

File: charts/tf-controller/Chart.yaml [hunk 2]
```diff
@@ -2,5 +2,5 @@ apiVersion: v2
 name: tf-controller
 description: The Helm chart for Weave GitOps Terraform Controller
 type: application
-version: 0.15.2
-appVersion: "v0.15.2"
+version: 0.15.3
```
Comment (chanwit) [inline charts/tf-controller/Chart.yaml:5]: Please do not dump the version yet. We haven't released v0.15.2.
It's OK if the chart test fails for now.

**Review by chanwit — CHANGES_REQUESTED** [review]

File: charts/tf-controller/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Weave GitOps Terraform Controller
 
-![Version: 0.15.2](https://img.shields.io/badge/Version-0.15.2-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v0.15.2](https://img.shields.io/badge/AppVersion-v0.15.2-informational?style=flat-square)
+![Version: 0.15.3](https://img.shields.io/badge/Version-0.15.3-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v0.15.3](https://img.shields.io/badge/AppVersion-v0.15.3-informational?style=flat-square)
```
Comment (chanwit) [inline charts/tf-controller/README.md:3]: Please bump version to v0.16.0-rc.1, as we'll ship the branch planner in this version.

**Review by chanwit — CHANGES_REQUESTED** [review]

File: charts/tf-controller/Chart.yaml (hunk 2 above)
Comment (yitsushi) [reply charts/tf-controller/Chart.yaml:5]: Sure I can, but why not a simple `0.15.x` patch? The only change in the helm chart is the `RUNTIME_NAMESPACE` addition to the branch planner + removed healthz endpoint references from branch planner (as it does not have any http endpoints at all).

Comment (yitsushi) [reply charts/tf-controller/Chart.yaml:5]: > Also, the branch planner is a big thing, we cannot use a point release for it.

it was already part of the chart.

Comment (yitsushi) [reply charts/tf-controller/Chart.yaml:5]: The bump script did nothing just shows the help of `yq` a few times. Tomorrow I'll check it and bump.

**Review by chanwit — APPROVED** [review]

**Pull request merged**